### DISCUSSION TITLE
How to run a single project test?

### DISCUSSION THREAD

**Discussion by lazka** [mesonbuild/meson#11640] [Q&A] [answered]
I want to just run `test cases\frameworks\15 llvm` to debug an error.

There is `python3 run_project_tests.py --only "frameworks"`

which brings it down to 35 tests, but that takes a long time.

The website at https://mesonbuild.com/Contributing.html suggests to run:

```
python3 meson.py test\ cases/frameworks/15\ llvm/ builddir
```

but that doesn't run the tests, at least it doesn't fail as with `run_project_tests.py`

**Comment by jpakkane** [comment]
This is probably because the test suite runs it multiple times with different values for options. See `test.json` in the directory for values.

**Comment by eli-schwartz** [comment, marked as answer]
You can run `--only frameworks/15` and it will narrow down the tests even further than just all frameworks. (Technically you can specify the full project name there. But as long as it's a unique prefix it will all work out. Note that "15" will run both 15 and 150-159, except that there aren't quite that many framework tests... It's worth keeping in mind for `--only common/15` though.)

It will still run the full matrix for the fifteenth frameworks project, which tbh is what you want anyway. That will be a lot fewer than 35 though. :)

**Reply by lazka** [reply]
I see thanks, so `python3 run_project_tests.py --only "frameworks/15 llvm"` works in my case.
